coresdk/src/backend/text_driver.cpp

```diff
@@ -131,7 +171,12 @@ void sk_draw_text(
         _sk_draw_bitmap_text(surface, x, y, text, clr);
         return;
     }
-    if (! (font && font->_data)) return; // error with font
+    if (! (VALID_PTR(font, FONT_PTR))) return; // error with font
+
+    // Check that the font size has been loaded and create if not
+    if (font->_data.count(font_size) == 0) {
+        font->_data[font_size] = TTF_OpenFont(font->filename.c_str(), font_size);
```

Comment:
Add in checks to see if this worked. raise warning if it fails and return without drawing.